src/tree/node_children.rs

```diff
@@ -560,33 +559,33 @@ mod inner {
         /// Access to the nodes array.
         #[inline(always)]
         pub fn nodes<'a>(&'a self) -> &'a [Arc<Node>] {
-            unsafe { mem::transmute(&self.nodes[..(self.len())]) }
+            unsafe { &*(&self.nodes[..(self.len())] as *const _ as *const _) }
         }
 
         /// Mutable access to the nodes array.
         #[inline(always)]
-        pub fn nodes_mut<'a>(&'a mut self) -> &'a mut [Arc<Node>] {
-            unsafe { mem::transmute(&mut self.nodes[..(self.len as usize)]) }
+        pub fn nodes_mut(&mut self) -> &mut [Arc<Node>] {
```

Comment:
This is possibly an odd stylistic preference, but I actually prefer the explicit lifetimes here, largely because of the unsafe stuff going on inside the function.  I realize it doesn't *actually* make a difference, but it makes me feel better.